applications/processor/http-request-processor/src/main/java/org/springframework/cloud/stream/app/processor/http/request/HttpRequestProcessorProperties.java

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright 2020-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.app.processor.http.request;
+
+import java.util.Map;
+
+import javax.validation.constraints.AssertTrue;
+import javax.validation.constraints.NotNull;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.cloud.fn.http.request.HttpRequestFunctionProperties;
+import org.springframework.cloud.fn.http.request.HttpRequestProperties;
+import org.springframework.expression.Expression;
+import org.springframework.expression.spel.standard.SpelExpressionParser;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.messaging.Message;
+import org.springframework.validation.annotation.Validated;
+
+/**
+ * Configuration properties for the Http Request Processor module.
+ *
+ * @author Waldemar Hummer
+ * @author Mark Fisher
+ * @author Christian Tzolov
+ * @author Artem Bilan
+ * @author David Turanski
+ */
+
+@Validated
+@ConfigurationProperties("http.request.processor")
+public class HttpRequestProcessorProperties implements HttpRequestProperties {
```

Comment:
Even though you don't create a function as a bean over here, you still can configure `http.request` properties.
So, I don't see anything wrong to have two these configuration classes on the processor and do respective logic to distribute properties combination. This way we indeed won't need a common interface to support and will avoid some confusion in the target projects.

Reply:
This is a consequence of the boot metadata/doc plugin. The only way to generate all the properties in the README AFAICT.  I tried some other (obvious) combinations.

Reply:
Also:
* The validation rules (e.g., only one of urlExpression, url) require the processor properties to "inherit" or compose with the function properties.  The validation rules are different for function properties (e.g. url is required).
* The same prefix for all properties in the app context is less confusing IMHO

Reply:
OK. Thanks. Resolved then!